### ISSUE TITLE
Special characters break JSON.stringify when downloading the route

### ISSUE THREAD

**Issue by RischDev** [RischDev/rischdev.github.io#14]
As shown with Lurking's BN5 route, using special characters will cause JSON stringify to break and not generate a full valid JSON route. This means that any work done after the special character is lost, and the notes cannot be properly used. 

Currently, only '#' is known to break JSON.stringify, but there may be other characters. If you are authoring a route, avoid using special characters as much as possible (or save backups before using them, and test them out first).

**Comment by RischDev** [comment]
According to TJ09, the '#' character would be parsed as a URL character, thus causing the generation of the JSON file to break. By using `btoa(JSON.stringify(route))`, the '#' character remains part of the JSON, and the remaining text is parsed correctly.

Verified the fix and deployed, still need to confirm if other special characters break the Generate JSON function.

**Comment by RischDev** [comment]
Turns out other characters did break the btoa function. Determined that certain characters (unknown which ones) caused an exception in btoa, which would prevent the function from generating the JSON file. Read more here: https://developer.mozilla.org/en-US/docs/Web/API/btoa#unicode_strings

Due to this, and the fact btoa is deprecated, I switched it over to using Buffer to convert the route into a base64 string. Read more here: https://stackoverflow.com/a/70733727

Going to mark this as resolved for now, as I have verified the issue no longer occurs with existing routes.
